src/api/app/controllers/webui/request_controller.rb

```diff
@@ -488,6 +490,16 @@ def set_action_id
     @action_id = params[:request_action_id] || @supported_actions.first&.id || @actions.first.id
   end
 
+  def set_bs_request_action
+    # Make sure the action belongs to the request
+    if @action_id.in?(@bs_request.bs_request_actions.ids)
+      @bs_request_action = BsRequestAction.find(@action_id)
```

Comment:
I believe you can just do `@bs_request.bs_request_actions.find(@action_id)` (untested)

Reply:
works like a charm, pushed